rpcs3/Emu/Cell/lv2/lv2.cpp

```diff
@@ -2171,7 +2172,35 @@ bool lv2_obj::wait_timeout(u64 usec, ppu_thread* cpu, bool scale, bool is_usleep
 #endif
 		// TODO: Tune for other non windows operating sytems
 
-		if (g_cfg.core.sleep_timers_accuracy < (is_usleep ? sleep_timers_accuracy_level::_usleep : sleep_timers_accuracy_level::_all_timers))
+		const sleep_timers_accuracy_level accuarcy_type = g_cfg.core.sleep_timers_accuracy;
+		const u64 avg_delay = get_avg_timer_reponse_delay();
+
+		static atomic_t<u64> g_success = 0;
+		static atomic_t<u64> g_fails = 0;
+
+		if (accuarcy_type == sleep_timers_accuracy_level::_dynamic && avg_delay < 30 && avg_delay < (remaining + 15) / 2)
+		{
+			wait_for(remaining);
+
+			if (remaining < host_min_quantum)
+			{
+				g_success += remaining;
+				//g_success++;
+			}
+
+			passed = get_system_time() - start_time;
```

Comment:
I'd rather not use system clock function here. It is notoriously slow by itself with that 128-bit time-stretched division. We're also evaluating real-time values using a time-stretching clock source, making the numbers weird.

Reply:
I disabled the 128 bit path.

Reply:
Keep in mind that there's still a faulty ifdef somewhere

Reply:
The problem is that you're comparing a stretched clock with a non-stretched quantum. That obviously makes the same values have different meanings depending on user configuration. e.g If I have my clock running twice as fast, the comparison against host quantum means we incorrectly use spinning instead of sleeping. Stuff like that.
get_system_time is abused quite a lot, it really should be limited to guest operations only and keep an unscaled clock for host interactions.

Reply:
Technically this line does not change anything because it was how all code paths were ended before too in the loop iteration. I only moved this line in this code path too. I rather not introduce new changes in this pr regarding this behavior.

Reply:
It was also wrong before. It's a problem all over CELL code is what I'm trying to say. If you're already overhauling how the timers work, why not fix it while you're at it?

Reply:
get_system_time returns host time, it is get_guest_system_time that returns guest time. The comparison against quantum here compares against host time, am I missing something?

Reply:
Oh, ok, I see now. I was mixing up the two because of the old 128-bit thing. This is correct then.